File: rift-runtime/src/Rift.Runtime/Commands/ArgAction.cs
// ===========================================================================
// Rift
// Copyright (C) 2024 - Present laper32.
// All Rights Reserved
// ===========================================================================

namespace Rift.Runtime.Commands;

/// <summary>
/// See https://docs.rs/clap/latest/clap/enum.ArgAction.html for more details. <br/>
///
/// Just make sure all sites are equivalent. <br/>
/// <remarks>
/// This is not clap! We just use subset of it.
/// </remarks>
/// </summary>
internal enum ArgAction
{
    Set,
    Append,
    SetTrue,
    SetFalse,
    Count,
    Help,
    HelpShort,
    HelpLong,
    Version,   
}

File: rift-cli/src/de/arg/arg_action.rs
use clap::ArgAction as ClapArgAction;
use serde::{Deserialize, Serialize};

enum_de!(ClapArgAction, ArgAction,
    #[derive(Deserialize, Serialize, Clone, Copy)]
    {
        Set,
        Append,
        SetTrue,
        SetFalse,
        Count,
        Help,
        HelpShort,
        HelpLong,
        Version,
    }
);
